src/components/tree/Tree.tsx

```diff
@@ -0,0 +1,152 @@
+import { useCallback, useEffect, useState } from 'react';
+
+import { useDebounce } from 'react-use';
+
+import { useFilters } from '@/hooks/useFilters';
+import { useToolbox } from '@/hooks/useToolbox';
+import { Collection, Link } from '@/typings/stac';
+
+import { TreeNode } from './TreeNode';
+import { filterTree } from './utils';
+
+import './Tree.scss';
+
+export type TreeCatalog = {
+  id: string;
+  title: string;
+  type: 'Catalog';
+  catalogs?: TreeCatalog[]; // sub-catalogs
+  collections?: Collection[]; // collections
+};
+
+// Utility function to fetch data from a given URL
+const fetchData = async (url: string) => {
+  const response = await fetch(url);
+  return response.json();
+};
+
+type TreeProps = {
+  catalogUrl: string;
+};
+
+export const Tree = ({ catalogUrl }: TreeProps) => {
+  const [treeData, setTreeData] = useState<TreeCatalog[]>([]);
+  const [expandedNodes, setExpandedNodes] = useState<{ [key: string]: boolean }>({});
+  const [filteredTreeData, setFilteredTreeData] = useState<TreeCatalog[]>([]);
+
+  const {
+    state: { activeFilters },
+  } = useFilters();
+
+  const {
+    actions: { setActivePage, setSelectedCollection },
+  } = useToolbox();
+
+  // Recursive function to fetch and build the tree structure
+  const fetchCatalog = useCallback(async (url: string): Promise<TreeCatalog> => {
+    const catalogData = await fetchData(url);
+
+    // Check if the current catalog has sub-catalogs or collections
+    const subCatalogs: TreeCatalog[] = await Promise.all(
+      catalogData.links
+        ?.filter((link) => link.rel === 'child')
```

Comment:
Is it true that child links always point to sub-catalogs?

Reply:
From what I saw yes, all `catalogs` were `rel=child` and all `collections` where `rel=data`. I looked at all catalogs we have harvested and couldn't see any that differed.

Reply:
I must be misunderstanding somewhere because the first catalog I looked at `airbus` has:
```
{
    "rel": "child",
    "type": "application/json",
    "href": "https://dev.eodatahub.org.uk/api/catalogue/stac/catalogs/supported-datasets/airbus/collections/airbus_sar_data"
}
```

Which is a collection

Reply:
Okay, I'll check on that, more likely I am wrong, you've been on the STAC stuff for a lot longer. Either that or it is a bug in the airbus collection. Let me look into this specifically.